.github/workflows/main.yml

```diff
@@ -32,6 +32,30 @@ jobs:
 
       - run: python mirror.py
 
+      - name: Check for unpushed commits
+        id: check_unpushed
+        run: |
+          UNPUSHED_COMMITS=$(git log origin/main..HEAD)
+          if [ -z "$UNPUSHED_COMMITS" ]; then
+            echo "No unpushed commits found."
+            echo "changes_exist=false" >> $GITHUB_ENV
+          else
+            echo "Unpushed commits found."
+            echo "changes_exist=true" >> $GITHUB_ENV
+          fi
+
       - run: |
           git push origin HEAD:refs/heads/main
           git push origin HEAD:refs/heads/main --tags
+        if: env.changes_exist == 'true'
```

Comment:
Can you move these to under the id or name field for clarity?

Reply:
Ah basically asking for this instead:

```
      - name: push changes if they exist
        if: env.changes_exist == 'true'
        run: |
          git push origin HEAD:refs/heads/main
          git push origin HEAD:refs/heads/main --tags
```

So it's not dangling under `run: |` which I find hard to read